back-end/src/sciler/config/workingConfigTypes.go

```diff
@@ -0,0 +1,320 @@
+	Name  string `json:"name"`
+	Rules []Rule `json:"rules"`
+}
+
+// GetName returns the name of a GeneralEvent
+func (g GeneralEvent) GetName() string {
+	return g.Name
+}
+
+// GetRules returns the rules of a GeneralEvent
+func (g GeneralEvent) GetRules() []Rule {
+	return g.Rules
+}
+
+// Event is an interface that both Puzzle and GeneralEvent implement
+type Event interface {
+	GetName() string
+	GetRules() []Rule
+}
+
+func compare(param1 interface{}, param2 interface{}, comparision string) bool {
+	switch comparision {
+	case "eq":
+		return reflect.DeepEqual(param1, param2)
+	case "lt":
+		return param1.(float64) < param2.(float64)
+	case "gt":
+		return param1.(float64) > param2.(float64)
+	case "lte":
+		return param1.(float64) <= param2.(float64)
+	case "gte":
+		return param1.(float64) >= param2.(float64)
+	case "contains":
+		return contains(param1, param2)
+	default:
+		panic(fmt.Sprintf("Cannot compare on: %s", comparision))
+	}
+}
+
+func contains(list interface{}, element interface{}) bool {
+	slice := reflect.ValueOf(list)
+	for i := 0; i < slice.Len(); i++ {
+		if element == slice.Index(i).Interface() {
+			return true
+		}
+	}
+	return false
+}
+
+// Device is a struct for all the devices
+type Device struct {
+	ID          string
+	Description string
+	Input       map[string]string
+	Output      OutputObject
+	Status      map[string]interface{}
+	Connection  bool
+}
+
+// Condition is a struct that determines when rules are fired.
+type Condition struct {
+	Type        string            `json:"type"`
+	TypeID      string            `json:"type_id"`
+	Constraints LogicalConstraint `json:"constraints"`
+}
+
+// CheckConstraints is a method that checks types and comparator operators
+func (condition Condition) CheckConstraints(config WorkingConfig) error {
+	return condition.Constraints.CheckConstraints(condition, config)
+}
+
+// Resolve is a method that checks if a condition is met
+func (condition Condition) Resolve(config WorkingConfig) bool {
+	return condition.Constraints.Resolve(condition, config)
+}
+
+// Constraint is main constraint type with comparison and componentID (latter only for device condition).
+type Constraint struct {
+	Comparison  string
+	ComponentID string
+	Value       interface{}
+}
+
+// CheckConstraints is a method that checks types and comparator operators
+func (constraint Constraint) CheckConstraints(condition Condition, config WorkingConfig) error {
+	switch condition.Type {
+	case "device":
+		{
+			if device, ok := config.Devices[condition.TypeID]; ok {
+
+				valueType := reflect.TypeOf(constraint.Value).Kind()
+				comparision := constraint.Comparison
+				if inputType, ok := device.Input[constraint.ComponentID]; ok {
+					switch inputType {
+					case "string":
+						{
+							if valueType != reflect.String {
+								return fmt.Errorf("input type string expected but %s found as type of value %v", valueType.String(), constraint.Value)
+							}
+							if comparision != "eq" {
```

Comment:
contains is also possible

Reply:
contains is not possible, maybe make `subset` a comparison of string when this is needed